server/xpub/entities/manuscript/resolvers.js

```diff
@@ -150,14 +150,24 @@ ${err}`,
 
       const pubsub = await getPubsub()
       let uploadedSize = 0
+      let startTime = process.hrtime()
       const progressReport = new Transform({
         transform(chunk, encoding, done) {
           uploadedSize += chunk.length
           const uploadProgress = Math.floor((uploadedSize * 100) / fileSize)
+          if (uploadProgress < 100) {
+            const elapsedTime = process.hrtime(startTime)
+            // MS_IN_S = 1e3, NS_IN_MS = 1e6
+            const elapsedTimeMs = elapsedTime[0] * 1e3 + elapsedTime[1] / 1e6
+            if (elapsedTimeMs < 500) {
+              return done(null, chunk)
+            }
+            startTime = process.hrtime()
+          }
           pubsub.publish(`${ON_UPLOAD_PROGRESS}.${user}`, {
```

Comment:
@dryhten  Maybe we also want to  throttle the messages to 500ms apart? ... this would prevent any flooding too?

Reply:
You mean in general for all websocket messages using something like a decorator or specifying this somewhere in config? Maybe the PR description was not good enough, but this waits for 500ms between each message, not just the first one.